### PR TITLE
OCPBUGS-34649: provisioning: add ownership for tls secret

### PR DESCRIPTION
**Pull request by vrutkovs**
Make sure all TLS artifacts have clear ownership

### CODE REVIEW

File: provisioning/baremetal_secrets.go [hunk 1]
```diff
@@ -194,6 +195,9 @@ func createOrUpdateTlsSecret(info *ProvisioningInfo) error {
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      tlsSecretName,
 			Namespace: info.Namespace,
+			Annotations: map[string]string{
+				annotations.OpenShiftComponent: "Bare Metal Hardware Provisioning / baremetal-operator ",
```
Comment (honza) [inline provisioning/baremetal_secrets.go:199]: nit: is the trailing space intentional?

File: provisioning/baremetal_secrets.go [hunk 2]
```diff
@@ -29,6 +30,7 @@ const (
 	inspectorSecretName      = "metal3-ironic-inspector-password"
 	inspectorUsername        = "inspector-user"
 	tlsSecretName            = "metal3-ironic-tls" // #nosec
+	baremetalJiraComponent   = "Bare Metal Hardware Provisioning / baremetal-operator"
```
Comment (dtantsur) [inline provisioning/baremetal_secrets.go:33]: I think we have a separate component for cluster-baremetal-operator

Comment (vrutkovs) [reply provisioning/baremetal_secrets.go:33]: not sure what's the difference between the two, but sure, lets make it cluster-baremetal-operator

Comment (dtantsur) [reply provisioning/baremetal_secrets.go]: Two unfortunately named and related but still quite different projects.

**Pull request merged**